share/spack/setup-env.fish

```diff
@@ -667,6 +667,19 @@ set -l sp_source_file (status -f)  # name of current file
 
 
 
+#
+# Identify and lock the python interpreter
+#
+for cmd in "$SPACK_PYTHON" python3 python python2
+    set -l _sp_python (command -s "$cmd")
```

Comment:
For consistency with POSIX shells, we should consider using `command -v` instead of `-s` here:
> For basic compatibility with POSIX command, the -v flag is recognized as an alias for -s
(fish `command` man-pages)

Reply:
I think we can do this one later since it's just in the `fish` script.